### PR TITLE
Fix/8435 fetch statistics for an entire federal state

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Add Federal state fetching in local statistics.
Now we have a new data structure called "Region" which can be a District or a federal State to avoid code duplication

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8435

## Screenshots
https://user-images.githubusercontent.com/15270737/126188933-387cbe11-7323-44d3-a2cd-a8bbf728389e.MP4

### CODE REVIEW

**Review by freshking — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift [hunk 1]
```diff
@@ -90,15 +89,70 @@ class HomeStatisticsTableViewCell: UITableViewCell {
...
+				onUpdate()
+			}
+			.store(in: &subscriptions)
+		
+		keyFigureCellModel.$localFederalStates
+			.receive(on: DispatchQueue.OCombine(.main))
+			.sink { localFederalStates in
+				Log.debug("update with \(keyFigureCellModel.localFederalStates.count) federal local stats", log: .localStatistics)
+				
+
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:134]:
```suggestion
				
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift [hunk 2]
```diff
@@ -113,46 +167,71 @@ class HomeStatisticsTableViewCell: UITableViewCell {
...
 				
 				self.removeLocalStatisticsCards()
-
+				
 				for singleSelectedLocalStatistics in selectedLocalStatistics {
+					
+					// check to separate the single entry and multiple entries
+					self.localStatisticsRegion = singleSelectedLocalStatistics.localStatisticsRegion
+					let regionName = singleSelectedLocalStatistics.localStatisticsRegion.name
+					let regionStatistics: SevenDayData
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:176]: Also a possible name: `RegionStatistics` or `RegionStatisticsData`

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift [hunk 3]
```diff
@@ -90,15 +89,70 @@ class HomeStatisticsTableViewCell: UITableViewCell {
...
 		self.cellModel = keyFigureCellModel
-
+		
 		keyFigureCellModel.$localAdministrativeUnitStatistics
 			.receive(on: DispatchQueue.OCombine(.main))
 			.sink { administrativeUnitsData in
-				Log.debug("update with \(keyFigureCellModel.localAdministrativeUnitStatistics.count) local stats", log: .localStatistics)
+				Log.debug("update with \(keyFigureCellModel.localAdministrativeUnitStatistics.count) administrative local stats", log: .localStatistics)
+				
+				// check to separate the single entry and multiple entries
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:98]: I have no idea what this means 😅

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift [hunk 4]
```diff
@@ -90,15 +89,70 @@ class HomeStatisticsTableViewCell: UITableViewCell {
...
+				
+				let administrativeUnit = administrativeUnitsData.first {
+					$0.administrativeUnitShortID == UInt32(self.localStatisticsRegion?.id ?? "0")
+				}
+				
+				// needed for UI updates
+				self.localStatisticsCache = store
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:105]: I don't understand, isn't this both just the store?

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift [hunk 5]
```diff
@@ -113,46 +167,71 @@ class HomeStatisticsTableViewCell: UITableViewCell {
 				Log.debug("update with \(keyFigureCellModel.selectedLocalStatistics.count) local stats", log: .localStatistics)
 				
 				self.removeLocalStatisticsCards()
-
+				
 				for singleSelectedLocalStatistics in selectedLocalStatistics {
+					
+					// check to separate the single entry and multiple entries
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:173]: Again no idea 🙈

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift (hunk 2 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:176]: you mean instead of SevenDayData?

**Review by nickguendling — APPROVED** [review]

**Review by KaiTeuber — APPROVED** [review]

**Pull request merged**